pandas/core/groupby.py

```diff
@@ -4437,20 +4437,15 @@ def _reorder_by_uniques(uniques, labels):
 
 
 def _groupby_indices(values):
+    """
+    return a dictionary of groups -> indices
+    """
+    if not is_categorical_dtype(values):
+        values = Categorical(values)
```

Comment:
we *are* re-using the existing factorization (if its categorical already its unchanged).